### PR TITLE
[AIDAPP-237]: Introduce automatic auditing on newly created data models

### PR DESCRIPTION
**Pull request by kandarp-canyon**
### Ticket(s) or GitHub Issue

- https://canyongbs.atlassian.net/browse/AIDAPP-237

### Technical Description

> Auto auditing has been implemented.

### Any deployment steps required?

> No.

### Are any Feature Flags Added?

>No.

_______________________________________________

### CODE REVIEW

File: app-modules/audit/src/Settings/AuditSettings.php
Comment (Orrison) [inline app-modules/audit/src/Settings/AuditSettings.php:1]: I believe the `HasDefaultSettings` interface and `defaults` method are not needed here and can be removed and deleted as we changed to a better default handling that works with the defaults of the parameter.

Try that and see if it solves the issue with the test. If not let me know and I can take a look further.

**Review by Orrison — CHANGES_REQUESTED** [review]

File: app-modules/audit/src/Settings/AuditSettings.php
Comment (kandarp-canyon) [reply app-modules/audit/src/Settings/AuditSettings.php:1]: Hi @Orrison , I've pushed the changes after removing the HasDefaultSettings interface and defaults method, but the test error is still there. can you please check?

Comment (kandarp-canyon) [reply app-modules/audit/src/Settings/AuditSettings.php:1]: @Orrison Please check and provide your valuable feedback.

File: app-modules/audit/tests/AuditListenerTest.php [hunk 1]
```diff
@@ -39,24 +39,26 @@
...
+    $auditSettings->audited_models_exclude = [$serviceRequest->getMorphClass()];
 
     $auditSettings->save();
 
-    $serviceRequest = ServiceRequest::factory()->create();
-
     expect($serviceRequest->audits)->toHaveCount(0);
 
-    $auditSettings->audited_models[] = $serviceRequest->getMorphClass();
+    $auditSettings->audited_models_exclude[] = [];
```
Comment (Orrison) [inline app-modules/audit/tests/AuditListenerTest.php:55]: This should be resetting it to a blank array, not setting a new value as a blank array

```suggestion
    $auditSettings->audited_models_exclude = [];
```

**Review by Orrison — CHANGES_REQUESTED** [review]

**Review by Orrison — APPROVED** [review]

**Pull request merged**